hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/ReplicationOptions.java

```diff
@@ -81,7 +81,21 @@ public void setReplication(String replication) {
   }
 
   // Option is defined in subclasses
-  public void setType(ReplicationType type) {
-    this.type = type;
+  public void setType(String type) {
+    try {
+      ReplicationType replicationType = ReplicationType.valueOf(type);
+      if (replicationType == ReplicationType.CHAINED
+              || replicationType == ReplicationType.STAND_ALONE) {
+        throw new IllegalArgumentException(
+                String.format("Unsupported replication type %s",
+                        replicationType.name()));
+      }
+      this.type = replicationType;
+    } catch (IllegalArgumentException ex) {
+      throw new IllegalArgumentException(
+              String.format("Invalid value '%s' for option '--type'. %s." +
+                      " Supported values are: RATIS, EC.",
+                      type, ex.getMessage()));
+    }
```

Comment:
The error message for unsupported replication type is now more cumbersome than previously:

```
Could not invoke public void org.apache.hadoop.ozone.freon.FreonReplicationOptions.setType(java.lang.String) with STAND_ALONE (java.lang.IllegalArgumentException: Invalid value 'CHAINED' for option '--type'. Unsupported replication type CHAINED. Supported values are: RATIS, EC.)
```

and

```
Could not invoke public void org.apache.hadoop.ozone.shell.ShellReplicationOptions.setType(java.lang.String) with asdf (java.lang.IllegalArgumentException: Invalid value 'asdf' for option '--type'. No enum constant org.apache.hadoop.hdds.client.ReplicationType.asdf. Supported values are: RATIS, EC.)
```

Ideally it should be:

```
Unsupported replication type CHAINED. Supported values are: RATIS, EC.
```

and

```
Unsupported replication type asdf. Supported values are: RATIS, EC.
```

So we should get rid of the message about invoking `setType`, and simplify the Ozone-specific part (which is due to two levels of `IllegalArgumentException`).

To achieve this, I think we may need a more complex change.  So I guess this one is OK as a temporary solution.

Reply:
Agree! thanks Attila!!